tests/src/Interop/COM/Activator/Servers/AssemblyA.cs

```diff
@@ -16,4 +17,100 @@ public object GetTypeFromC()
     {
         return this._fromC.GetType();
     }
+}
+
+public class ValidRegistrationTypeCallbacks : IValidateRegistrationCallbacks
+{
+    [ComRegisterFunctionAttribute]
+    public static void RegisterFunction(Type t) => s_didRegister = true;
+
+    [ComUnregisterFunctionAttribute]
+    public static void UnregisterFunction(Type t) => s_didUnregister = true;
+
+    private static bool s_didRegister = false;
+    private static bool s_didUnregister = false;
+
+    bool IValidateRegistrationCallbacks.DidRegister() => s_didRegister;
+
+    bool IValidateRegistrationCallbacks.DidUnregister() => s_didUnregister;
+
+    void IValidateRegistrationCallbacks.Reset()
+    {
+        s_didRegister = false;
+        s_didUnregister = false;
+    }
+}
+
+public class ValidRegistrationStringCallbacks : IValidateRegistrationCallbacks
+{
+    [ComRegisterFunctionAttribute]
+    public static void RegisterFunction(string t) => s_didRegister = true;
+
+    [ComUnregisterFunctionAttribute]
+    public static void UnregisterFunction(string t) => s_didUnregister = true;
+
+    private static bool s_didRegister = false;
+    private static bool s_didUnregister = false;
+
+    bool IValidateRegistrationCallbacks.DidRegister() => s_didRegister;
+
+    bool IValidateRegistrationCallbacks.DidUnregister() => s_didUnregister;
+
+    void IValidateRegistrationCallbacks.Reset()
+    {
+        s_didRegister = false;
+        s_didUnregister = false;
+    }
+}
+
+public class NoRegistrationCallbacks : IValidateRegistrationCallbacks
```

Comment:
We should have test for multiple register and unregister functions on a class. Whatever the desired behavior is, we should test it.

Reply:
Sure, I can add that but if memory serves method enumeration is undefined so I don't really know how to test something that has nor defined order. Is there a way I can force that?

Reply:
Now that we are throwing this is easy.